File: greedy/1196. How Many Apples Can You Put into the Basket/Solution.cs
public class Solution {
    public int MaxNumberOfApples(int[] weight) {
            
        Array.Sort(weight);
        int remainingCapacity = 5000;
        int count = 0;

        foreach (int appleWeight in weight) {
            if (appleWeight > remainingCapacity)
                break;
                
            remainingCapacity -= appleWeight;
            count++;
        }
        
        return count;
    }
}

File: greedy/1196. How Many Apples Can You Put into the Basket/solution.py
class Solution:
    def maxNumberOfApples(self, weight: List[int]) -> int:
        cumulative_weights = list(accumulate(sorted(weight)))
        
        for i, total_weight in enumerate(cumulative_weights):
            if total_weight > 5000:
                return i
        
        return len(weight)
